pkg/compactor/block_upload.go

```diff
@@ -65,7 +67,29 @@ func (c *MultitenantCompactor) StartBlockUpload(w http.ResponseWriter, r *http.R
 		return
 	}
 
-	if err := c.createBlockUpload(ctx, r, logger, userBkt, tenantID, blockID); err != nil {
+	content, err := io.ReadAll(http.MaxBytesReader(w, r.Body, maximumMetaSizeBytes))
+	if err != nil {
+		if errors.As(err, new(*http.MaxBytesError)) {
+			err = httpError{
+				message:    fmt.Sprintf("The request body was too large (maximum size allowed is %d bytes)", maximumMetaSizeBytes),
```

Comment:
instead of "the request body", we can be more specific: "meta.json file" or "block metadata".

Reply:
Changed it to "block metadata".